lib/workers/repository/dependency-dashboard.ts

```diff
@@ -220,15 +241,36 @@ function getBranchesListMd(
   if (filteredBranches.length === 0) {
     return '';
   }
-  let result = `## ${title}\n\n${description}\n\n`;
-  result += `${filteredBranches
-    .map((branch: BranchConfig): string => getListItem(branch, listItemType))
-    .join('')}`;
+
+  const { categories, uncategorized } =
+    splitBranchesByCategory(filteredBranches);
+  let result = `## ${title}\n\n${description}`;
+  if (Object.keys(categories).length > 0) {
+    for (const [category, branches] of Object.entries(categories).sort(
+      ([keyA], [keyB]) => keyA.localeCompare(keyB),
+    )) {
+      result = result.trimEnd() + '\n\n';
+      result += `### ${category}\n\n`;
+      result += getBranchList(branches, listItemType);
+    }
+    if (uncategorized.length > 0) {
+      result = result.trimEnd() + '\n\n';
+      result += `### Others`;
+    }
+  }
+  result = result.trimEnd() + '\n\n';
+  result += getBranchList(uncategorized, listItemType);
+
   if (bulkComment && bulkMessage && filteredBranches.length > 1) {
+    if (Object.keys(categories).length > 0) {
+      result = result.trimEnd() + '\n\n';
+      result += '### All\n\n';
+    }
     result += getCheckbox(bulkComment);
-    result += `${bulkIcon ? bulkIcon + ' ' : ''}**${bulkMessage}**${bulkIcon ? ' ' + bulkIcon : ''}\n`;
+    const icon = ` ${bulkIcon ?? ''} `;
+    result += `${icon}**${bulkMessage}**${icon}`.trim();
```

Comment:
why trim? also newline is renoved

Reply:
To delete the leading and trailing space we got from `icon`. I prefered this over:
```suggestion
    result += `${bulkIcon ? bulkIcon + ' ' : ''}**${bulkMessage}**${bulkIcon ? ' ' + bulkIcon : ''}`;
```